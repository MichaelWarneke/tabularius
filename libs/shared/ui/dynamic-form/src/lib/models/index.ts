import {
  FormControl,
  ValidatorFn,
  AbstractControlOptions,
  AsyncValidatorFn,
  FormArray
} from '@angular/forms';

export class FormControlBase extends FormControl {
  controlType = '';
  key = '';
  label = '';
  maxWidth = 100;
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlTextbox extends FormControlBase {
  controlType = 'input';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlDate extends FormControlBase {
  controlType = 'datepicker';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlCheckbox extends FormControlBase {
  controlType = 'checkbox';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlSelect extends FormControlBase {
  controlType = 'select';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlRadioButton extends FormControlBase {
  controlType = 'radio-button';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}
export class FormControlArray extends FormArray {
  controlType = 'array';
  key = '';
  label = '';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}
export { Validators } from '@angular/forms';
