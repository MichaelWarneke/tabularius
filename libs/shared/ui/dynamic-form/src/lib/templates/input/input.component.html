<mat-form-field [formGroup]="group">
  <input matInput placeholder="{{val?.key}}" [formControlName]="val?.key" />
</mat-form-field>
