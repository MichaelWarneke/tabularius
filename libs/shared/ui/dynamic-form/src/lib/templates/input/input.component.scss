.tabu-input {
  padding: 10px;
  width: 40%;
}
