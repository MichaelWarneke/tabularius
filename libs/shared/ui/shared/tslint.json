{
  "extends": "../../../../tslint.json",
  "rules": {
    "directive-selector": [true, "attribute", "tabuSharedUi", "camelCase"],
    "component-selector": [true, "element", "tabu-shared-ui", "kebab-case"]
  }
}
