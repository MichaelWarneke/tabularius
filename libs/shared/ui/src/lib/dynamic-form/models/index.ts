export { QuestionBase } from './question-base';
export { DropdownQuestion } from './question-dropdown';
export { TextboxQuestion } from './question-textbox';
export { DatepickerQuestion } from './question-datepicker';
