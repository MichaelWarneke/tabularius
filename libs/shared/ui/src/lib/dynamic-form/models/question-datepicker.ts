import { QuestionBase } from './question-base';

export class DatepickerQuestion extends QuestionBase<string> {
  controlType = 'datepicker';
  type: Date;

  constructor(options: any = {}) {
    super(options);
    this.type = options['type'] || '';
  }
}
