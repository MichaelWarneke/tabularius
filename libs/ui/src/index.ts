export { TabuAppSidenavModule } from './lib/nav';
