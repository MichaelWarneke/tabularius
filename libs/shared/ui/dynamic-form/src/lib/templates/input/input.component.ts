import { Component, OnInit } from '@angular/core';
import { FormGroup } from '@angular/forms';
import { FormControlTextbox } from '../../models';
import { BaseComponent } from '../base.component';

@Component({
  selector: 'tabu-dynamic-form-input',
  templateUrl: './input.component.html',
  styleUrls: ['./input.component.scss']
})
export class InputComponent extends BaseComponent {
  constructor() {
    super();
  }
}
