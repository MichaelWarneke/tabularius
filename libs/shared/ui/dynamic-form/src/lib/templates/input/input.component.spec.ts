import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { InputComponent } from './input.component';
import { SharedModule } from '@tabularius/shared/ui';
import { FormGroup, FormControl } from '@angular/forms';
import { FormControlTextbox } from '../../models';
import { By } from '@angular/platform-browser';

describe('InputComponent', () => {
  let component: InputComponent;
  let fixture: ComponentFixture<InputComponent>;
  const key = 'testKey';
  const label = 'testLabel';

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule, NoopAnimationsModule],
      declarations: [InputComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(InputComponent);
    component = fixture.componentInstance;
    component.group = new FormGroup({ testKey: new FormControl('') });
    component.control = new FormControlTextbox();
    component.control.key = key;
    component.control.label = label;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should contain the label', () => {
    const attributes = fixture.debugElement.query(By.css('input')).attributes;
    const text = attributes['placeholder'];
    expect(text).toContain(label);
  });

  it('should contain the formControlName in attributes', () => {
    const attributes = fixture.debugElement.query(By.css('input')).attributes;
    const text = attributes['ng-reflect-name'];
    expect(text).toContain(key);
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });
});
