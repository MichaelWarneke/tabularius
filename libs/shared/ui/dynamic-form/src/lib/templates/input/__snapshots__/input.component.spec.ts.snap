// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponent should render 1`] = `
<tabu-dynamic-form-input
  control={[Function FormControlTextbox]}
  group={[Function FormGroup]}
>
  <mat-form-field
    class="tabu-input mat-form-field ng-tns-c3-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
    ng-reflect-form="[object Object]"
  >
    <div
      class="mat-form-field-wrapper"
    >
      <div
        class="mat-form-field-flex"
      >
        
        
        <div
          class="mat-form-field-infix"
        >
          <input
            aria-invalid="false"
            aria-required="false"
            class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
            id="mat-input-3"
            matinput=""
            ng-reflect-name="testKey"
            ng-reflect-placeholder="testLabel"
            placeholder="testLabel"
          />
          <span
            class="mat-form-field-label-wrapper"
          >
            
            <label
              aria-owns="mat-input-3"
              class="mat-form-field-label ng-tns-c3-0 mat-empty mat-form-field-empty ng-star-inserted"
              for="mat-input-3"
              id="mat-form-field-label-7"
              ng-reflect-disabled="true"
              ng-reflect-ng-switch="false"
            >
              
              
              testLabel
              
              
            </label>
          </span>
        </div>
        
      </div>
      
      <div
        class="mat-form-field-underline ng-tns-c3-0 ng-star-inserted"
      >
        <span
          class="mat-form-field-ripple"
        />
      </div>
      <div
        class="mat-form-field-subscript-wrapper"
        ng-reflect-ng-switch="hint"
      >
        
        
        <div
          class="mat-form-field-hint-wrapper ng-tns-c3-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
          style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
        >
          
          <div
            class="mat-form-field-hint-spacer"
          />
        </div>
      </div>
    </div>
  </mat-form-field>
</tabu-dynamic-form-input>
`;
